packages/web-components/fast-foundation/src/accordion-item/accordion-item.template.ts

```diff
@@ -17,9 +21,10 @@ export function accordionItemTemplate(
                 role="heading"
                 aria-level="${x => x.headinglevel}"
             >
+                ${startSlotTemplate(options)}
                 <button
-                    class="button"
-                    part="button"
+                    class="control"
+                    part="control"
```

Comment:
What's our new semantic for "control" here? 

Previously, it was essentially used in scenarios like button, inputs, radio, checkbox where the host *does not* represent the control, an element within it does. So, for checkbox, the Shadow DOM includes the label, thus the *control* is the portion without that. The accordion item itself includes the heading, I think in this case, the entire thing is the control (unlike the checkbox example).

Reply:
I tried to keep 'control' as the general name for the main thing you interact with. In many components we used a more specific name for that core thing, like 'toolbar' or 'radiogroup'. I was looking for a stronger rule here.

'Accordion item' breaks convention from the other similar components like 'combobox' or 'select', and in this case I think it's fair that 'heading' is actually the whole thing. The reason it's strange is because we stretch the button over the whole heading. I'm not sure that's always the case or why we originally chose to implement it that way rather than putting the event on the 'heading'.

For the "input" class of components, I used 'control' to indicate the logical container for the input. It was previously 'root' but I thought that was misleading because it was not the root or base of the component.

Reply:
In the "input" components I called the `<input />` "input", so perhaps in aligning with calling things what they are (toolbar, radiogroup, etc) we call the internal button and anchor, "button" and "anchor". That at least leaves "control" for something where I think it's well-served like the input case. I think we should have one common word for the thing you interact with on those components, other than "root".

Naming is hard.

Reply:
This sort of consistency and convention is crucial for the upcoming playwright tests.

Reply:
I think the concern I have with some of these is that the host element in many cases *is* the thing. It's not the underlying bit, the whole thing is the toolbar, radiogroup, "input". Specifically for inputs, if we change that to be something which represents part of the input but is not *actually* an input, we end up inconsistent again. Part of me with that has concerns around scale because the fact that its an input rather than a *div* is internal and exposing that via parts can send an odd message.

Reply:
In the case of "input", it seems material that the "input" is actually an `<input/>`. It's the place where the value is going to be entered, and there are specific considerations for styling an `<input/>` that in structure is nested in other elements that provide the visual boundary.

Here I think the concern is the "label". There's a "label" and there's "something else".
If that "something else" is "input" because it logically represents the visual boundary of where the `value` goes, that's fine, but it may be confusing that it's actually a container `<div>` and not an `<input/>`.
If that "something else" is "control" because it's a general name for the part of a `component` that you interact with, I'm not sure that holds up well for things that aren't typically considered "controls". I think the "tooltip" and "toolbar" fall into this category. "Component" works well because it can represent anything, "control" historically represents more of the `form` category of `button`s and `input`s.

I don't see a clear recommendation as to the name we use here? From the Playwright perspective are we saying that fewer top-level names is better, like for that reason everything should be "control" if that at all makes sense?

The strongest consistent pattern I see right now is that we always use the name of the thing. For instance, "toolbar" = "toolbar", "text field" = "text field", and "button" = "button". The current button template seems like exactly the same case where the fact that we happen to use a `<button>` is not material because that could also be a `<div>` with a bunch of events hooked up to it. Do we think that calling that `<div>` "control" is the most consistent because it represents the thing you interact with? What about the case where "control" feels wrong because it's not interactive (tooltip, etc.)

Reply:
> In the case of "input", it seems material that the "input" is actually an <input/>. It's the place where the value is going to be entered, and there are specific considerations for styling an <input/> that in structure is nested in other elements that provide the visual boundary.

I'd again argue that this is just an implementation detail. This could easily end up converting a div with content editable and the semantics moving to the host with a role of textbox. From a form association standpoint, the entire control is the "input" or form element, we've just decided to leverage native elements and focus delegation. Were that to change, the semantic fails because it's not an input and we need to rename it according to the proposed principle of naming things what they are. Were we to name this based on a principle of _intent_ or _purpose_ then a more generic name has more opportunity to scale. It doesn't mean it always will, but the principle is less brittle.

> "Component" works well because it can represent anything, "control" historically represents more of the form category of buttons and inputs.

The _component_ is the whole thing - in this case, text field includes a label. Same with radio, checkbox, combobox. The internals are not "the thing", the host element is the thing and it includes those things. Specifically with Toolbar, the label is part of the toolbar, they aren't two separate things. The positioning region within toolbar exists to position the slotted items relative to the label. So it could be inline with the items, it could be above them, but it's within the toolbar and it's all part of the toolbar. Were we to remove the label from that (or radiogroup as it also has the same intent) that part would no longer be needed and the entire thing would flatten. A good example is what you've done here with dialog, where we had a positioning region to ensure that we could position without hitting odd blur issues when transforming absolutely positioned elements. Now that the host can serve that purpose, it's no longer needed and the whole thing flattens out. So the semantic of `positioning region` scales across all of these consistently in a way that when someone sees it they understand why it's there and the intent. I'm not saying that we have to stick with it, but I think it shows that we can have a consistent semantic that can be used across all of our controls where once it's used it can be understood consistently. This is the case for those building components and those using components. Once I see and understand the intent, I can more quickly create components that follow the same semantic. As a consumer of the library, because I have a consistent part name, I can understand its use across all components.